src/backend/drm/surface/atomic.rs

```diff
@@ -856,6 +856,11 @@ impl<A: AsRawFd + 'static> AtomicDrmSurface<A> {
 
         // and finally the others
         for plane_info in planes {
+            req.add_property(
+                plane_info.handle,
+                plane_prop_handle(&*prop_mapping, plane_info.handle, "CRTC_ID")?,
+                property::Value::CRTC(Some(self.crtc)),
+            );
```

Comment:
I don't see anything wrong with this, but, why is the primary plane handled separately. It would seem you could add primary to the planes slice and kill off some duplicated code.

Reply:
Yes, I did this as part of #760 I just wanted to keep the scope of this PR cleaner.